lib/auth.js

```diff
@@ -231,6 +234,47 @@ module.exports = new Model({
     }.bind(this));
   },
 
+  signInWithOAuthRedirect: function (args, window) {
```

Comment:
Overriding `window` seems like a bad thing. I also think it's reasonable to expect this to run in a certain context (the browser).